common/Environments/Models/ComputeSystem.cs

```diff
@@ -275,6 +275,126 @@ public async Task<ComputeSystemOperationResult> ConnectAsync(string options)
         }
     }
 
+    public async Task<ComputeSystemOperationResult> PinToStartMenuAsync(string options)
+    {
+        var localErrorString = $"PinToStartMenuAsync for: {this} failed due to exception";

```

Comment:
nit: Declare when it's needed. It's only needed for logging under catch. No need to use it in "throw" since it goes into traces anyway. And then no need to declare it at all since it can be inlined in _log.Error.